recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,266 @@
+        self._cmake.definitions["NV_FORCE_64BIT_SUFFIX"] = False
+        self._cmake.definitions["NV_FORCE_32BIT_SUFFIX"] = False
+        self._cmake.definitions["PX_ROOT_LIB_DIR"] = os.path.abspath(os.path.join(self.package_folder, "lib")).replace("\\", "/")
+
+        if self.settings.os == "Windows":
+            # Options defined in physx/source/compiler/cmake/windows/CMakeLists.txt
+            self._cmake.definitions["PX_COPY_EXTERNAL_DLL"] = False # External dll copy disabled, PhysXDevice dll copy is handled afterward during conan packaging
+            self._cmake.definitions["PX_FLOAT_POINT_PRECISE_MATH"] = self.options.enable_float_point_precise_math
+            self._cmake.definitions["PX_USE_NVTX"] = False # Could be controlled by an option if NVTX had a recipe, disabled for the moment
+            self._cmake.definitions["GPU_DLL_COPIED"] = True # PhysXGpu dll copy disabled, this copy is handled afterward during conan packaging
+
+            # Options used in physx/source/compiler/cmake/windows/PhysX.cmake
+            self._cmake.definitions["PX_GENERATE_GPU_PROJECTS"] = False
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _get_physx_build_type(self):
+        if self.settings.build_type == "Debug":
+            return "debug"
+        elif self.settings.build_type == "RelWithDebInfo":
+            return "checked"
+        elif self.settings.build_type == "Release":
+            if self.options.release_build_type == "profile":
+                return "profile"
+            else:
+                return "release"
+
+    def _get_target_build_platform(self):
+        return {
+            "Windows" : "windows",
+            "Linux" : "linux",
+            "Macos" : "mac",
+            "Android" : "android",
+            "iOS" : "ios"
+        }.get(str(self.settings.os))
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), self._get_license())
+
+        out_lib_dir = os.path.join(self.package_folder, "lib", self._get_physx_build_type())
+        self.copy(pattern="*.a", dst="lib", src=out_lib_dir, keep_path=False)
+        self.copy(pattern="*.so", dst="lib", src=out_lib_dir, keep_path=False)
+        self.copy(pattern="*.dylib*", dst="lib", src=out_lib_dir, keep_path=False)
+        self.copy(pattern="*.lib", dst="lib", src=out_lib_dir, keep_path=False)
+        self.copy(pattern="*.dll", dst="bin", src=out_lib_dir, keep_path=False)
+
+        tools.rmdir(out_lib_dir)
+        tools.rmdir(os.path.join(self.package_folder, "source"))
+
+        self._copy_external_bin()
+
+    def _get_license(self):
+        readme = tools.load(os.path.join(self._source_subfolder, "README.md"))
+        begin = readme.find("Copyright")
+        end = readme.find("\n## Introduction", begin)
+        return readme[begin:end]
+
+    def _copy_external_bin(self):
```

Comment:
This method is copying prebuilt binaries.
Are these binaries absolutely necessary? Isn't this recipe/source release building them?
On Windows, I removed these files in the `bin` directory, and test_package kept working..

Reply:
It's required if consumer initiates a PxCudaContextManager and end user has cuda. If end user doesn't have cuda, physx switch back to cpu and it doesn't even try to load these binaries, so there is no error. But if end user has cuda and these binaries are missing, application crashes I guess (I can check again on my personnal desktop, tried this few months ago).
Source code of these binaries is not available.

Reply:
Yeah, I noticed a fallback to software simulator on my system.
But it seemed my problem were the drivers that were too old.

It's too ad that that the source is not available,but expected from nvidea.

Just asking, but are these binaries drivers or developer files?
I mean, what can be expected from users/developers: use these libraries or those from the gpu driver?

Reply:
I quote PhysX documentation ;)

From https://github.com/NVIDIAGameWorks/PhysX/blob/4.1/physx/documentation/platformreadme/windows/readme_windows.html:
> The PhysXDevice/64.dll must be redistributed with GPU-enabled applications, and must live in the PhysXGpu.dll directory.

From https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/BuildingWithPhysX.html#generating-projects-with-cmake:

> Redistribution
>
>On the Windows platform, you need to redistribute some of our DLLs to end users as part of your application:
>
>    - PhysXCommon_*.dll - will always be needed.
>    - PhysX_*.dll - will always be needed.
>    - PhysXFoundation_*.dll - will always be needed.
>    - PhysXCooking_*.dll - you only need to bundle if your application cooks geometry data on the fly.
>    - PhysXGPU_*.dll - is only needed if your application runs some simulation on the GPU.
>
>Where * is a platform specific suffix, e.g. 32 or 64. You will need one or the other depending on whether your application is built in 64 bit mode.

Reply:
Haha, these links are much appreciated!
Other os'es are not mentioned. I assume they are in a similar situation?

Reply:
GPU part of PhysX is not supported for others os (except for Linux 64 bits).
For Linux 64 bits, PhysXDevice doesn't exist, only PhysXGPU. There is no explicit instructions about this shared lib in their linux documentation (https://github.com/NVIDIAGameWorks/PhysX/blob/4.1/physx/documentation/platformreadme/linux/readme_linux.html).

Did you look to this source file?
https://github.com/NVIDIAGameWorks/PhysX/blob/4.1/physx/source/physx/src/gpu/PxPhysXGpuModuleLoader.cpp

Reply:
Thanks for clarification!
It makes sense now.
Maybe add a small comment about the why/how for Windows and Linux?

Reply:
comments added